assignment/core/src/main/java/com/assignment/core/crypto/AESCipherProvider.java

```diff
@@ -0,0 +1,58 @@
+package com.assignment.core.crypto;
+
+import org.osgi.service.component.annotations.Component;
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+@Component(
```

Comment:
The implementations should be in a `impl` subpackage as per convention

Reply:
done